client/state/comments/reducer.js

```diff
@@ -0,0 +1,238 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import Immutable from 'immutable';
+
+/**
+ * Internal dependencies
+ */
+import {
+	COMMENTS_RECEIVE,
+	COMMENTS_REMOVE_COMMENT,
+	COMMENTS_ERROR_COMMENT,
+	COMMENTS_COUNT_RECEIVE,
+	COMMENTS_REQUEST,
+	COMMENTS_REQUEST_SUCCESS,
+	COMMENTS_REQUEST_FAILURE
```

Comment:
@gwwar I've made some [initial implementation](https://github.com/Automattic/wp-calypso/pull/3143/files#diff-b3d7f2f424cf387123e9eb8f231cf85aR333) without validating schema, tell me what you think.